src/Core/Checkout/Cart/Order/OrderConverter.php

```diff
@@ -307,6 +327,16 @@ public function assembleSalesChannelContext(OrderEntity $order, Context $context
             SalesChannelContextService::VERSION_ID => $context->getVersionId(),
         ];
 
+        $delivery = $order->getPrimaryOrderDelivery();
+
+        if (!Feature::isActive('v6.8.0.0')) {
+            $delivery = $order->getDeliveries()?->first();
+        }
+
+        if ($delivery !== null) {
+            $options[SalesChannelContextService::SHIPPING_METHOD_ID] = $delivery->getShippingMethodId();
+        }
+
```

Comment:
you don't seem to use the $delivery here 🤔

Reply:
or in other words, you also set the shippingMethodId below, one of them seems redundant